client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -84,30 +120,36 @@ private final class PoolManager[F[_], A <: Connection[F]](
         if (!isClosed) {
           @tailrec
           def go(): Unit =
-            idleQueue.dequeueFirst(_.requestKey == key) match {
+            getConnectionFromQueue(key) match {
               case Some(conn) if !conn.isClosed =>
                 logger.debug(s"Recycling connection: $stats")
                 callback(Right(NextConnection(conn, fresh = false)))
 
               case Some(closedConn) =>
                 logger.debug(s"Evicting closed connection: $stats")
-                allocated -= 1
+                decrConnection(key)
                 go()
 
-              case None if allocated < maxTotal =>
+              case None if numConnectionsCheckHolds(key) =>
                 logger.debug(s"Active connection not found. Creating new one. $stats")
                 createConnection(key, callback)
 
-              case None if idleQueue.nonEmpty =>
+              case None if curTotal == maxTotal =>
                 logger.debug(
```

Comment:
We should move the log message into the two respective branches since it would be nice to know if we ended up killing a random idle connection or added it to the wait queue.

Reply:
Ok